### DISCUSSION TITLE
[Feature Request - Logs Visualization] Copy/Paste log entry as JSON

### DISCUSSION THREAD

**Discussion by cyrille-leclerc** [grafana/grafana#61482] [Observability]
When troubleshooting a performance problem, I want to capture the details of a log entry copying all its details (incl timestamp and labels) as JSON.
Existing copy button is just about the log message, omitting the log entry labels and timestamp.

I would like this GUI to be consistent with:
* https://github.com/grafana/grafana/issues/61391

![image](https://user-images.githubusercontent.com/459691/212078100-df0cc545-dfef-4d1e-854e-6e54280e2159.png)

**Comment by cyrille-leclerc** [comment]
Duplicate of:
* https://github.com/grafana/grafana/discussions/61371
Hase been converted into GH issue: https://github.com/grafana/grafana/issues/61393

**Comment by stevendungan** [comment]
ty @cyrille-leclerc ! notes from our brief discussion:

* problem to solve is collaborative troubleshooting; the idea is users could put this information in e.g. GH issue, JIRA ticket, Slack message
* the shortcoming today is with the copy button just producing a log message is that it doesn't give enough context to effectively reproduce the issue - without labels and timestamp (key ingredients in a performant loki search), it is difficult to query Loki to recreate the log line.

constraints:
* the output should ideally be standardized to align with Tempo's output if they were to fulfill the same request (Jenny Lam's team); I think we'd need to validate what that could look like and make sure it works
* my immediate thought is that two copy buttons, or a "copy text/json" or "copy message/line" toggle, would be a crowded User Interface, though team may be able to think through more creatively - meaning we would want to choose one or the other

Finally, we have an alternative solution we discussed which is a [permalink](https://github.com/grafana/grafana/discussions/56093), which has been requested before. The advantage of this approach is fewer clicks/actions to a result; the disadvantage is it doesn't allow for the key content of the log message to be included in the e.g. JIRA ticket, which may be a key need for users to action on that ticket.

_Edit_: Another thing we discussed that would be helpful no matter what was seeing if we could get some qual/quant validation on how or if the feature is used today, and what those users think.

**Reply by stevendungan** [reply]
@cyrille-leclerc wondering if the permalink idea + the copy output as is would actually solve this problem - "here is the offending log message, and here is the permalink to recreate" without needing to even change the copy behavior?

**Reply by cyrille-leclerc** [reply]
Thanks Steven, I suspect that the capability to copy all the details of a log message will be useful but we can wait for the equivalent for traces to be implemented:
* https://github.com/grafana/grafana/issues/61391
